jooqdb/src/main/java/org/cristalise/storage/jooqdb/JooqClusterStorage.java

```diff
@@ -1,339 +1,426 @@
+
+                domainHandlers.add( (JooqDomainHandler) Class.forName(handlerClass).newInstance());
+            }
+        }
+        catch (InstantiationException | IllegalAccessException | ClassNotFoundException ex) {
+            Logger.error("JooqClusterStorage could not instantiate domain handler");
+            Logger.error(ex);
+            throw new PersistencyException("JooqClusterStorage could not instantiate domain handler:"+ex.getMessage());
+        }
+
+        if (! Gateway.getProperties().getBoolean(JOOQ_DISABLE_DOMAIN_CREATE, false)) {
+            context.transaction(nested -> {
+                for (JooqDomainHandler handler: domainHandlers) handler.createTables(DSL.using(nested));
+            });
+        }
+    }
+
+    public void dropHandlers() throws PersistencyException {
+        DSLContext context = JooqHandler.connect();
+        for (JooqHandler handler: jooqHandlers.values()) handler.dropTables(context);
+    }
+
+    @Override
+    public void close() throws PersistencyException {
+        Logger.msg(1, "JooqClusterStorage.close()");
+        try {
+            JooqHandler.connect().close();
+        }
+        catch (Exception e) {
+            Logger.error(e);
+            throw new PersistencyException(e.getMessage());
+        }
+    }
+
+    @Override
+    public void postBoostrap() throws PersistencyException {
+        JooqHandler.connect().transaction(nested ->{
+            for (JooqDomainHandler domainHandler : domainHandlers) domainHandler.postBoostrap(DSL.using(nested));
+        });
+        
+        if(Gateway.getProperties().containsKey(JOOQ_AUTOCOMMIT)){
+
+            Gateway.getProperties().put(JOOQ_AUTOCOMMIT, false);
+            HikariConfig config = new HikariConfig();
+            JooqHandler.ds.copyStateTo(config);
+            config.setAutoCommit(false);
+            config.addDataSourceProperty( "autoCommit",false);
+            HikariDataSource newDs = new HikariDataSource(config);    
+            JooqHandler.ds = newDs;
+           
+        }
+    }
+
+    @Override
+    public void postStartServer() throws PersistencyException {
+        JooqHandler.connect().transaction(nested ->{
+            for (JooqDomainHandler domainHandler : domainHandlers) domainHandler.postStartServer(DSL.using(nested));
+        });
+    }
+
+    @Override
+    public void postConnect() throws PersistencyException {
+        if(Gateway.getProperties().containsKey(JOOQ_AUTOCOMMIT)){
+            Gateway.getProperties().put(JOOQ_AUTOCOMMIT, true);
+        }
+        JooqHandler.connect().transaction(nested ->{
+            for (JooqDomainHandler domainHandler : domainHandlers) domainHandler.postConnect(DSL.using(nested));
+        });
+        
+    }
+
+    @Override
+    public void begin(Object locker) {
+        try {
+            DSLContext context = JooqHandler.connect();
+            contextMap.put(locker, context);

```

Comment:
Why do we put it in the Map even if locker is null?

Reply:
Changes will be on next commit.